### PR TITLE
spine upgrade 3.8

### PR DESCRIPTION
**Pull request by sunnylanwanjun**
jsb-adapter: https://github.com/cocos-creator-packages/jsb-adapter/pull/185
cocos2d-x-lite:https://github.com/cocos-creator/cocos2d-x-lite/pull/1885
fireball:https://github.com/cocos-creator/fireball/pull/9158

### CODE REVIEW

**Review by SantyWang — APPROVED** [review]

File: extensions/spine/editor/spine-meta.js [hunk 1]
```diff
@@ -135,6 +141,10 @@ class SpineMeta extends CustomAssetMeta {
     }
 
     static validate (assetpath) {
+        // handle binary file
+        if (assetpath.indexOf(".skel") > 0) {
```
Comment (jareguo) [inline extensions/spine/editor/spine-meta.js]: 感觉这里用 endsWith 会更直观

File: extensions/spine/editor/spine-meta.js [hunk 2]
```diff
@@ -178,32 +215,44 @@ class SpineMeta extends CustomAssetMeta {
...
+
+            this._initTexture(asset, fspath, cb);
         });
     }
+
+    import (fspath, cb) {
+        if (fspath.indexOf(".skel") > 0) {
+            this._importBinary(fspath, cb);
+        } else {
+            CustomAssetMeta.prototype.import.call(this, fspath, cb);
```
Comment (jareguo) [inline extensions/spine/editor/spine-meta.js]: ES6 class 可以用 this.super(fspath, cb);

File: extensions/spine/editor/spine-meta.js [hunk 3]
```diff
@@ -178,32 +215,44 @@ class SpineMeta extends CustomAssetMeta {
...
-                db.saveAssetToLibrary(this.uuid, asset);
-                cb();
-            });
+            this._initTexture(asset, fspath, cb);
+        });
+    }
+
+    _importBinary (fspath, cb) {
+        // import native asset
+        let extname = Path.extname(fspath).toLowerCase(); // 注意，这里会把所有后缀统一改成小写
```
Comment (jareguo) [inline extensions/spine/editor/spine-meta.js]: 开源仓库还是用英文注释吧

File: extensions/spine/editor/spine-meta.js [hunk 4]
```diff
@@ -178,32 +215,44 @@ class SpineMeta extends CustomAssetMeta {
...
+        let dest = this._assetdb._uuidToImportPathNoExt(this.uuid) + extname;
+        Fs.copy(fspath, dest, err => {
+            if (err) {
+            return cb(err);
```
Comment (jareguo) [inline extensions/spine/editor/spine-meta.js]: 缩进优化下~

**Review by jareguo — COMMENTED** [review]
汗…… Spine 真是越来越大了。Spine 官方没提供剔除二进制支持的方法吗？

**Comment by sunnylanwanjun** [comment]
> 汗…… Spine 真是越来越大了。Spine 官方没提供剔除二进制支持的方法吗？

你是说把spinebinary单独成一个模块？

File: extensions/spine/editor/spine-meta.js (hunk 2 above)
Comment (sunnylanwanjun) [reply extensions/spine/editor/spine-meta.js]: image

**Comment by jareguo** [comment]
> 你是说把spinebinary单独成一个模块？

算了…… Spine 资源变小后优化效果更明显

**Comment by balazsnemeth** [comment]
@sunnylanwanjun , @jareguo , Here it says: "Upgrade spine to 3.8". 

Can you let me know which is the latest version of Spine supported? v3.8.73? Or 3.8.55? We checked, and it is not the latest v3.8.95 for sure, and I cannot see any more details to figure it out, can you please help us on this?

**Comment by PatriceJiang** [comment]
3.8.55

**Pull request merged**